tests/test_camera.py

```diff
@@ -56,8 +59,13 @@ def test_can_see_something(self):
         camera = self.robot.cameras[0]
         tokens = camera.see()
 
-        # Check the correct markers are spotted
-        self.assertEqual({x.id for x in tokens}, {9})
+        self.assertEqual({x.id for x in tokens}, {9}, "Saw wrong markers")
+
+        self.assertEqual(
+            9,
+            tokens[0].id,
```

Comment:
Assuming we follow the standard of _not_ using [yoda conditions](https://en.wikipedia.org/wiki/Yoda_conditions), the first 2 args here should be reversed.

Reply:
Unfortunately unittest is rather unclear about this in its docs. While I agree with the yoda conditions comment, my experience is that the standard among unit testing frameworks is to have signatures like `assertEqual(expected, actual, message=None, *message_args)`.

This _is_ something which `unittest` conforms to, though you can only see it some of the time. Here's an example where it matters:
``` python
import unittest

class ExampleTest(unittest.TestCase):
    def test_exmaple(self):
        expected = {'expected': 'bar'}
        actual = {'actual': 'spam'}
        self.assertEqual(expected, actual)
```
which outputs:
```
F
======================================================================
FAIL: test_exmaple (goo.ExampleTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/sourcebots/robot-api/goo.py", line 7, in test_exmaple
    self.assertEqual(expected, actual)
AssertionError: {'expected': 'bar'} != {'actual': 'spam'}
- {'expected': 'bar'}
+ {'actual': 'spam'}

----------------------------------------------------------------------
Ran 1 test in 0.001s

FAILED (failures=1)
```
note the diff-like output of the expected vs actual.

If you put the expected and the actual the other way around, the message that you get would be rather confusing. I've unfortunately had a number of occasions (and quite a lot of time spent) where I've found that I'm investigating a failing test that I can't understand the reason for the failure only to find that the arguments to the assertion were back to front leading to incorrect output.

I therefore strongly prefer putting the expected first, even though a bit yoda it does read.

Reply:
I'm not sure how i've missed that after all this time... I know `assertIn` works like that, but as does the `in` operator, so i'm ok with that. 

Seeing as `==` works either way like this, I can live with it differing